File: android_fragment/app/src/main/java/com/support/generator/module/main/MainActivityModule.java
package com.support.generator.module.main;

import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 * @author: leellun
 * @data: 2018/12/24.
 */
@Module
public class MainActivityModule {

  @Provides
  public MainActivityContract.IMainPresenter provideMainPresenter() {
    return new MainActivityPresenter();
  }
}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/module/main/MainActivityModule.kt
package gauges.mobile.github.com.mvpapplication.module.main

import android.content.Context
import dagger.Module
import dagger.Provides
import gauges.mobile.github.com.mvpapplication.common.mvphelper.PresenterLoader

/**
 * @author: leellun
 * @data: 2018/12/19.
 *
 */
@Module
class MainActivityModule {

    @Provides
    fun providePresenterLoader(context: Context, presenter: MainActivityPresenter)
            : PresenterLoader<MainActivityContract.IMainPresenter> = PresenterLoader(context, presenter)
}